packages/peregrine/lib/hooks/useWindowSize.js

```diff
@@ -51,9 +53,10 @@ export const WindowSizeContextProvider = props => {
     // This hook has side effects of adding listeners so we only want to create it
     // once and store it in context for reference by components.
     const windowSize = useWindowSizeListener();
+    const isDesktop = windowSize.innerWidth >= WINDOW_SIZE_DESKTOP;
```

Comment:
Unfortunately, I don't think `useWindowSize` should evaluate the window size (which is why it doesn't do so already). Layout is presentation, so component code shouldn't know about layout, and that means component code shouldn't use window size to conditionally render elements. Breakpoint conditions belong in CSS.

- I realize we already have a couple of components that call `useWindowSize`, compare the result to some magic breakpoint number, and conditionally render elements based on that comparison. Ideally, they wouldn't do that; at minimum, let's not encourage more components to do so.

- I also realize that the whole reason for this PR is to move the conditional from CSS to component code in order to avoid excessive DOM size. Again, it would be great if we would not have to do that.

Have we considered any other ways to avoid rendering DOM elements? For example, could we use [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) to toggle a flag in state, which could then serve as the conditional for rendering elements?

Reply:
I was under the impression we were okay to use the window size since it was used elsewhere.
I will get back to the task and investigate how we can change that.
Ideally, we would remove the call to window size everywhere so others don't get caught like I was.

Reply:
@jimbo I've reverted all work on the useWindowSize hook and created the useIsInViewport hook instead to achieve similar results like you've proposed.